runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -741,8 +741,14 @@ ObsIdType __catalyst__qis__HamiltonianObs(MemRefT_double_1d *coeffs, int64_t num
 
 RESULT *__catalyst__qis__Measure(QUBIT *wire, int8_t postselect)
 {
+    std::optional postselectOpt = postselect;
+
+    // Absence of postselect is denoted now by a nullopt
+    if (postselect != 0 && postselect != 1) {
+        postselectOpt = std::nullopt;
+    }
```

Comment:
Ok so I don't know why yet, but the error on macOS only reproduces in release mode not in debug mode. I tried the following pattern (initialize on known safe rather than revert on known bad) because it seemed safer to me and it does get rid of the error, but it doesn't explain why there are bad values to begin with 👀

```suggestion
    std::optional<int8_t> postselectOpt;

    // Absence of postselect is denoted now by a nullopt
    if (postselect == 0 || postselect == 1) {
        postselectOpt = postselect;
    }
```

Reply:
@dime10, this is actually my fix here [c037545](https://github.com/PennyLaneAI/catalyst/pull/504/commits/c0375451ce618482f11883f564408f3419c7a534) and [7956f21](https://github.com/PennyLaneAI/catalyst/pull/504/commits/7956f21845b409f1e4601b26ade84731bb9bc410)

```
    if (postselect != 0 && postselect != 1) {
        postselectOpt = std::nullopt;
    }
```

And I could build the wheels successfully with this patch. check here:

https://github.com/PennyLaneAI/catalyst/actions/runs/7848922913/job/21421213973?pr=504

Reply:
This line used to be as follows,

```
    // Absence of postselect is denoted now by a nullopt
    if (postselect == -1) {
```

and surprisingly `postselect == -1` fails on only macOS Arm platforms. Still, don't have any good answer that why this is failed there?!

Reply:
I would try changing the i8 to i64 just to see if it has something to do with the calling convention.

Reply:
Also where is `int8_t` typedef being imported? Don't there need to be a `stdint` include?

Reply:
@erick-xanadu good point!

Reply:
@erick-xanadu The include is in the `types.h` file.

But it seems you are right and mac doesn't play well with the i8 as an integer type. Both i64 and i32 seems to fix the issue without any additional changes.

Reply:
[> First, parameters smaller than 32 bits are extended to 32-bit values in a manner consistent with their type: Signed types are sign-extended and unsigned types are zero-extended.](https://devblogs.microsoft.com/oldnewthing/20210621-00/?p=105327)

I would need to look into the IR for both the caller and the callee. But this seems entirely possible.

Reply:
I was wondering if it was related to sign extension because in Ali's test when converted to `int` it prints as 255. 
The fact that it only appears in release mode would also indicate that compiler optimizations might be to blame 🤔

Reply:
> But it seems you are right and mac doesn't play well with the i8 as an integer type

https://news.ycombinator.com/item?id=18269886

i8 seems to be an alias for an 'unsigned' char instead of 'signed' char on ARM processors.

I believe that updating the support to use i64 or i32 instead is a good decision.

Reply:
Interesting, I have changed all references to **postselect** to i32. Done.